packages/main/src/SegmentedButtonItem.ts

```diff
@@ -104,9 +117,73 @@ class SegmentedButtonItem extends ToggleButton implements ISegmentedButtonItem {
 	@property({ validator: Integer, defaultValue: 0 })
 	sizeOfSet!: number;
 
+	/**
+	 * Defines the text of the component.
+	 *
+	 * **Note:** Although this slot accepts HTML Elements, it is strongly recommended that you only use text in order to preserve the intended design.
+	 * @public
+	 */
+	@slot({ type: Node, "default": true })
+	text!: Array<Node>;
+
+	static i18nBundle: I18nBundle;
+
 	get ariaDescription() {
 		return SegmentedButtonItem.i18nBundle.getText(SEGMENTEDBUTTONITEM_ARIA_DESCRIPTION);
 	}
+
+	constructor() {
+		super();
+	}
+
+	_onclick() {
+		this.selected = !this.selected;
+	}
+
+	onEnterDOM() {
+		if (isDesktop()) {
+			this.setAttribute("desktop", "");
+		}
+	}
+
+	_onkeyup(e: KeyboardEvent) {
+		if (isSpaceShift(e)) {
+			e.preventDefault();
+		}
+	}
+
+	_ontouchend(e: TouchEvent) {
```

Comment:
Why do we need only ontouchend?

Reply:
I don't remember tbh(almost a month since i did the change), but removing it doesn't seem to break the control, so I will remove it.